labelme/widgets/canvas.py

```diff
@@ -490,10 +492,27 @@ def selectShapePoint(self, point, multiple_selection_mode):
 
     def calculateOffsets(self, shape, point):
         rect = shape.boundingRect()
-        x1 = rect.x() - point.x()
-        y1 = rect.y() - point.y()
-        x2 = (rect.x() + rect.width() - 1) - point.x()
-        y2 = (rect.y() + rect.height() - 1) - point.y()
+        left, right, top, bottom = (
+            rect.left(),
+            rect.right(),
+            rect.top(),
+            rect.bottom(),
+        )
+        for s in self.selectedShapes:
+            rect = s.boundingRect()
+            if rect.left() < left:
+                left = rect.left()
+            if rect.right() > right:
+                right = rect.right()
+            if rect.top() < top:
+                top = rect.top()
+            if rect.bottom() > bottom:
+                bottom = rect.bottom()
```

Comment:
I'm still not sure why you first iterate through `shape` and integrate through `self.selectedShapes` in `self.calculateOffsets()`.

Reply:
Is this nested for-loop really necessary?

Reply:
Try select more then one shape and move with that by mouse ... In one direction it will be outside the pixmap becuse self.calculateOffsets() in original version calculates only with one shape not all that selected ... this creates new virtual rectangle.

Reply:
I updated `self.calculateOffsets()`. Does that look good to you?